qiime2/sdk/proxy.py

```diff
@@ -22,7 +26,8 @@ def _create_proxy(self, selector):
         qiime_type = self._signature_[selector].qiime_type
```

Comment:
You could probably pass `qiime_type` into the proxy constructors instead of `self._signature_` since each proxy is always just a single entry.